lib/auth/bot.go

```diff
@@ -392,7 +383,41 @@ func (a *Server) generateInitialBotCerts(
 		generation:    generation,
 		loginIP:       loginIP,
 		botName:       botName,
-		botInstanceID: botInstanceID,
+	}
+
+	if experiment.Enabled() {
+		if existingInstanceID == "" {
+			// If no existing instance ID is known, create a new one.
+			uuid, err := uuid.NewRandom()
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			bi := newBotInstance(&machineidv1pb.BotInstanceSpec{
+				BotName:    botName,
+				InstanceId: uuid.String(),
+			}, initialAuth, expires.Add(machineidv1.ExpiryMargin))
+
+			_, err = a.BotInstance.CreateBotInstance(ctx, bi)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			certReq.botInstanceID = uuid.String()
+		} else {
+			// Otherwise, reuse the existing instance ID, and pass the
+			// initialAuth along.
+
+			// Note: botName is derived from the provision token rather than any
+			// value sent by the client, so we can trust it.
+			// TODO: Determine how we should handle update failure in edge cases
```

Comment:
I feel like we have two options here:

- Fail hard on the RPC side and mitigate this failure within `tbot`. This sounds a little more complex than I'd first think since `tbot` would have to create a new client w/o it's credentials and then rejoin.
- Create a new bot instance and then return certificates with that bot instance ID. 

I think generally, I like the idea of handling this server-side by just returning a new bot instance (as if none had been specified at all). If we design `tbot` around the idea that it's bot instance ID can change at the auth server's whim, then we shouldn't have a problem. I think this also nicely handles scenarios where someone has torn down their cluster, and restored it via Terraform or similar, the bot/token would exist, but the bot instances wouldn't. WDYT?

Reply:
I'm generally inclined to agree, I think the critical security control is that we don't ever reissue an instance ID unless we're certain it's the same bot; this wouldn't impact that at all, so I think it's fine.

I've added some logic to create a new instance here if the requested ID isn't found, and added a testcase for this behavior.